### ISSUE TITLE
Default value for gauges

### ISSUE THREAD

**Issue by robinwkurtz** [fluent/fluent-plugin-prometheus#211]
Hello, thanks for your plugin!

I'm currently using the gauge type for sensors such as temperature, humidity, and CO2. Unfortunately, if my process is not running to get these values my fluentd /metrics endpoint shows zero values for theme, since this endpoint is being scrapped by Prometheus, I now have zero values in my graphs. I wanted to remove these, but of course, in dealing with these types of sensors, a zero value valid so I cannot simply filter out "0" on Prometheus' side.

Is there a way to have the metrics endpoint show null values?

**Comment by Watson1978** [comment]
Thank you for using the plugin!

In IoT/sensor monitoring, "0 degrees" is a valid data point, not an indicator of "missing data."

However, due to the specification of the Prometheus exposition format, it is not possible to output a `null` value.
A metric either has a float value, or it must be completely absent.
Additionally, the underlying Ruby Prometheus client initializes gauges with a default value of 0 and retains the last known value until updated.

Instead of trying to output `null`, the standard Prometheus approach is to use a separate "health" metric and filter your data in PromQL (Grafana).

For example, if you can export a metric like `sensor_up = 1` (when running) and `sensor_up = 0` (when down), you can write your PromQL like this:

```
sensor_temperature_celsius and (sensor_up == 1)
```

This query will drop the temperature metric completely (making it effectively "null" or absent in the graph) whenever the sensor is down, preventing those fake "0 degree" dips.

In other words, you will need to devise a workaround on the sensor side (the data source) before the metrics reach Fluentd.

I hope this helps you achieve the graph you want! Since we cannot change the Prometheus format to support null, I will close this issue for now.
